ros/api/v1/hosts.py

```diff
@@ -148,9 +148,14 @@ def get(self):
                     row.PerformanceProfile.performance_utilization
                 )
                 host['idling_time'] = row.PerformanceProfile.idling_time
-                host['os'] = f"{row.System.operating_system['name']} " \
-                             f"{row.System.operating_system['major']}." \
-                             f"{row.System.operating_system['minor']}"
+                if (
+                        row.System.operating_system
+                        and all(key in row.System.operating_system.keys() for key in ['name', 'major', 'minor'])
+                ):
+                    host['os'] = f"{row.System.operating_system.get('name')} " \
+                                 f"{row.System.operating_system.get('major')}." \
+                                 f"{row.System.operating_system.get('minor')}"
+
```

Comment:
It seems that this logic will be required more than one place. Will it be good idea to encapsulate into new readonly property on System class? What do you think?

Reply:
@kgaikwad Changes :rocket: